components/ws-daemon/nsinsider/main.go

```diff
@@ -251,6 +252,162 @@ func main() {
 						return err
 					}
 
+					return nil
+				},
+			},
+			{
+				Name:  "setup-pair-veths",
+				Usage: "set up a pair of veths",
+				Flags: []cli.Flag{
+					&cli.IntFlag{
+						Name:     "target-pid",
+						Required: true,
+					},
+					&cli.StringFlag{
+						Name:     "name",
+						Required: true,
+					},
+				},
+				Action: func(c *cli.Context) error {
+					containerIf, vethIf, cethIf := "eth0", "veth0", "ceth0"
+					mask := net.IPv4Mask(255, 255, 255, 0)
+					vethAddr := net.IPNet{
+						IP:   net.IPv4(10, 0, 5, 1),
+						Mask: mask,
+					}
+					cethAddr := net.IPNet{
+						IP:   net.IPv4(10, 0, 5, 2),
+						Mask: mask,
+					}
+					masqueradeAddr := net.IPNet{
+						IP:   vethAddr.IP.Mask(mask),
+						Mask: mask,
+					}
+					ipCmd, iptablesCmd := "/usr/sbin/ip", "/usr/sbin/iptables"
+					netns := "workspace-ns"
+
+					cmd := exec.Command(ipCmd, "link", "add", vethIf, "type", "veth", "peer", "name", cethIf)
```

Comment:
@utam0k why not use "github.com/vishvananda/netlink"
```
veth := &netlink.Veth{
    LinkAttrs: netlink.LinkAttrs{
        Name:  vethIf,
        Flags: net.FlagUp,
        MTU:   1500,
        PeerName: cethIf,
    },
    PeerNamespace: netlink.NsFd(int(<hostNS>.Fd())),
}
```

to remove any package dependency in the workspace?

Reply:
It is just what I was looking for! Thanks. I redid the implementation.